framework/src/Volo.Abp.MongoDB/Volo/Abp/Domain/Repositories/MongoDB/MongoDbRepository.cs

```diff
@@ -131,6 +163,61 @@ public MongoDbRepository(IMongoDbContextProvider<TMongoDbContext> dbContextProvi
             return entity;
         }
 
+        public override async Task UpdateManyAsync(IEnumerable<TEntity> entities, bool autoSave = false, CancellationToken cancellationToken = default)
+        {
+            var isSoftDeleteEntity = typeof(ISoftDelete).IsAssignableFrom(typeof(TEntity));
+
+            foreach (var entity in entities)
+            {
+                SetModificationAuditProperties(entity);
+
+                if (isSoftDeleteEntity)
+                {
+                    SetDeletionAuditProperties(entity);
+                    await TriggerEntityDeleteEventsAsync(entity);
+                }
+                else
+                {
+                    await TriggerEntityUpdateEventsAsync(entity);
+                }
+
+                await TriggerDomainEventsAsync(entity);
+
+                SetNewConcurrencyStamp(entity);
+            }
+
+            if (BulkOperationProvider != null)
+            {
+                await BulkOperationProvider.UpdateManyAsync(this, entities, autoSave, cancellationToken);
+                return;
+            }
+
+            var entitiesCount = entities.Count();
+            BulkWriteResult result;
+
+            if (SessionHandle != null)
+            {
+                result = await Collection.BulkWriteAsync(SessionHandle, GetReplaceRequests());
+            }
+            else
+            {
+                result = await Collection.BulkWriteAsync(GetReplaceRequests());
+            }
+
+            if (result.MatchedCount < entitiesCount)
+            {
+                ThrowOptimisticConcurrencyException();
+            }
+
+            IEnumerable<WriteModel<TEntity>> GetReplaceRequests()
```

Comment:
Instead of a local method, just create the collection before `if (SessionHandle != null)` line and use inside the if/else blocks. Keep it simple unless we have an advantage of using local methods.

Reply:
Can you check Commit f3722985